### PR TITLE
Update model render items on texture load and fix no haze issue

### PR DESCRIPTION
**Pull request by SamGondelman**
https://highfidelity.manuscript.com/f/cases/11053/Transparent-materials-render-opaque

When a model overlay/entity is created, its URL changes, or its texture URLs change, we must wait until the textures load and then trigger a render update to properly mark the render items as opaque/transparent.

Also, in cases where there was no haze stage, we would not bind anything, so the buffer would be full of junk, instead of the default "off" values.

Test plan:
- Use teleport.  The model overlay at the end should render correctly.
- Find a model with transparent textures (there are plenty in the Volcano Lair content set - I used the light fixtures above the submarine).  It should render correctly.
- Reload content several times.  The above steps should still work after each reload.
- Go to an empty location.  You should not see any haze.
- From the empty location, create a model entity with transparent parts and create a zone around it with haze set to inherit.  Inside the zone, you should not see any haze and the object should render correctly.
- Change the zone's haze property to on.  You should see haze.
- Change it to off.  You should not see haze.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/5950/app_12091_PR.apk

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/5950/HighFidelity-Beta-PR12091-6a19b0e0b6f6a9d1fca8d67712cbbd63a730ed78.exe

**Comment by SamGondelman** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/5956/app_12091_PR.apk

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/5956/HighFidelity-Beta-PR12091-6a19b0e0b6f6a9d1fca8d67712cbbd63a730ed78.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/5956/HighFidelity-Beta-PR12091-6a19b0e0b6f6a9d1fca8d67712cbbd63a730ed78.dmg

**Comment by SamGondelman** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/5961/app_12091_PR.apk

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/5971/app_12091_PR.apk

**Comment by marko8904** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/5979/app_12091_PR.apk

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/5979/HighFidelity-Beta-PR12091-eb0d09360a4db1f809931c948821e91742bf27dc.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/5979/HighFidelity-Beta-PR12091-eb0d09360a4db1f809931c948821e91742bf27dc.dmg

**Comment by SamGondelman** [comment]
build this please

File: libraries/entities-renderer/src/RenderableModelEntityItem.cpp [hunk 1]
```diff
@@ -1080,6 +1080,10 @@ bool ModelEntityRenderer::needsRenderUpdate() const {
             return true;
         }
 
+        if (!_texturesLoaded && _model->getGeometry() && _model->getGeometry()->areTexturesLoaded()) {
```
Comment (samcake) [inline libraries/entities-renderer/src/RenderableModelEntityItem.cpp]: I think you need to use model and not-model here to make sure the pointer is valid

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/5995/app_12091_PR.apk

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/5995/HighFidelity-Beta-PR12091-a9d5dda97643bc5cd859be658aba9439fcc72f29.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/5995/HighFidelity-Beta-PR12091-a9d5dda97643bc5cd859be658aba9439fcc72f29.dmg

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/6004/app_12091_PR.apk

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/6004/HighFidelity-Beta-PR12091-5cf25a20c974333857a764477acb332956d68107.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/6004/HighFidelity-Beta-PR12091-5cf25a20c974333857a764477acb332956d68107.dmg

**Comment by QAadahlin** [comment]
Currently testing

**Comment by QAadahlin** [comment]
@conklin94122 Testing complete. PASSED.

**Pull request merged**